packages/terra-disclosure-manager/src/DisclosureManager.jsx

```diff
@@ -0,0 +1,416 @@
+     * executed in stack order. Components will be dismissed in order up until a rejection occurs, at which point
+     * the blocking component will be presented.
+     */
+    return this.resolveDismissChecksInSequence(disclosureKeys).then(() => {
+      this.dismissChecks = {};
+      this.closeDisclosure();
+    });
+  }
+
+  /**
+   * Creates an instance of a pop function for the component represented by the given key.
+   */
+  generatePopFunction(key) {
+    return () => {
+      let promiseRoot = Promise.resolve();
+
+      const dismissCheck = this.dismissChecks[key];
+      if (dismissCheck) {
+        promiseRoot = dismissCheck();
+      }
+
+      return promiseRoot
+        .then(() => {
+          this.dismissChecks[key] = undefined;
+          this.resolveDismissPromise(key);
+        })
+        .then(() => {
+          this.popDisclosure();
+        });
+    };
+  }
+
+  renderContentComponents() {
+    const { children, app } = this.props;
+
+    const appDelegate = {};
+
+    /**
+     * The disclose function provided will open the disclosure with the provided content.
+     */
+    appDelegate.disclose = (data) => {
+      if (this.disclosureTypeIsSupported(data.preferredType)) {
+        return this.safelyCloseDisclosure()
+          .then(() => {
+            this.openDisclosure(data);
+            /**
+             * The disclose Promise chain is resolved with a set of APIs that the disclosing content can use to
+             * manipulate the disclosure, if necessary.
+             */
+            return {
+              /**
+               * The afterDismiss value is a deferred Promise that will be resolved when the disclosed component is dismissed.
+               */
+              afterDismiss: new Promise((resolve) => {
+                this.onDismissResolvers[data.content.key] = resolve;
+              }),
+              /**
+               * The dismissDisclosure value is a function that the disclosing component can use to manually close the disclosure.
+               * Any and all dismiss checks are still performed.
+               */
+              dismissDisclosure: this.safelyCloseDisclosure,
+            };
+          });
+      }
+      return app.disclose(data);
+    };
+
+    return React.Children.map(children, child => React.cloneElement(child, {
+      app: AppDelegate.clone(app, appDelegate),
+    }));
+  }
+
+  renderDisclosureComponents() {
+    const { app } = this.props;
+    const { disclosureComponentKeys, disclosureComponentData, disclosureIsMaximized, disclosureIsFocused, disclosureSize } = this.state;
+
+    return disclosureComponentKeys.map((componentKey, index) => {
+      const componentData = disclosureComponentData[componentKey];
+      const isFullscreen = disclosureSize === availableDisclosureSizes.FULLSCREEN;
+      const popContent = this.generatePopFunction(componentData.key);
+
+      const appDelegate = {};
```

Comment:
Consider renaming this or `app` so it's easier to make sense of the madness in this method.